src/dask_awkward/lib/optimize.py

```diff
@@ -77,15 +78,15 @@ def optimize(
     if dask.config.get("awkward.optimization.enabled"):
         which = dask.config.get("awkward.optimization.which")
         if "columns" in which:
-            dsk = optimize_columns(dsk)
+            dsk = optimize_columns(dsk, keys)
         if "layer-chains" in which:
             dsk = rewrite_layer_chains(dsk, keys)
 
     return dsk
 
 
 def _prepare_buffer_projection(
-    dsk: HighLevelGraph,
+    dsk: HighLevelGraph, keys: Sequence[Key]
```

Comment:
I *believe* the keys from `__dask_optimize__` are always sequence of `Key`? So, I think we can impose that constraint on our own APIs. LMK if not and you'd prefer to support single keys too. Need to figure out how to do that, because the `Key` type is so general.

Reply:
I'm not sure if this is an assumption we can make. See dask.dataframe's [https://github.com/dask/dask/blob/bb9aa197b851726caa47250a3a12598a34249008/dask/dataframe/optimize.py#L15-L18](implementation). We can always coerce keys into a sequence -- which is what dataframe does